src/background.js

```diff
@@ -159,7 +159,14 @@ const messageOps = {
   },
   update: message => {
     Metrics.updateSnoozedTab(message);
-    return messageOps.cancel(message.old).then(() => messageOps.confirm(message.updated));
+    return removeAlarms(idForItem(message.old))
+      .then(() => {
+        const toSave = {};
+        toSave[idForItem(message.updated)] = message.updated;
+        return saveAlarms(toSave);
+      })
+      .then(updateWakeAndBookmarks)
+      .catch(reason => log('confirm rejected', reason));
```

Comment:
If the remove succeeds, and then the save fails, we'll lose the snoozed tab, which seems suboptimal.
Could we do this in the reverse order instead, so that the failure case will leave two snoozed tabs instead of none?